### PR TITLE
Create units summary table

### PR DESCRIPTION
**Pull request by abbiefarr**
## Issue
https://github.com/CityOfDetroit/bloom/issues/238

Addresses # (issue)

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Creates a UnitsSummary entity and dto, and the corresponding migration.

**Caveat**: I wasn't really sure how to structure the dto, would love some feedback there.

The table structure is described [here](https://github.com/bloom-housing/bloom/issues/1588#issuecomment-892034445), plus a `property` column which holds a `property.id` foreign key.

Current primary keys for the new table are `unitType`, `property`, and `monthlyRent`, but I expect we may want to add more fields to the composite key.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Run the migration and view the newly created table. I manually added a few rows to verify the primary keys.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

**Comment by abbiefarr** [comment]
I think the lint failure (403) could be fixed by adding `pull_request_target` to Bloom's lint.yml, based on [these docs](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target).

**Comment by abbiefarr** [comment]
@seanmalbert this is ready for another set of eyes, could someone on your team take a look?

File: backend/core/src/units-summary/dto/units-summary.dto.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { OmitType } from "@nestjs/swagger"
+import { UnitsSummary } from "../entities/units-summary.entity"
+
+export class UnitsSummaryDto extends OmitType(UnitsSummary, [] as const) {}
+export class UnitsSummaryCreateDto extends OmitType(UnitsSummaryDto, [] as const) {}
+export class UnitsSummaryUpdateDto extends OmitType(UnitsSummaryDto, [] as const) {}
```
Comment (seanmalbert) [inline backend/core/src/units-summary/dto/units-summary.dto.ts:6]: I think these can be fairly basic. If you take a look at `unit-types/dto/unit-type.dto.ts`, you can follow what was done there.

**Review by seanmalbert — COMMENTED** [review]
@abbiefarr ,
This is looking good. Please see my comment regarding the dto, it should be straight forward. Can you also please add an entry into the changelog for this work. 

One thing I'd like to think about, which I don't think we should cover here, is how do we foresee working with the service? In our initial case, we'll call it in the listing service on listing upsert, so we shouldn't need a separate controller for it. I imagine we'll want to do similar logic with directly saving the unit summary. So on upsert, we would check if the user has selected to use individual units to compute the unit summary or to save the unit summary directly. Does that make sense to you?

File: backend/core/src/units-summary/dto/units-summary.dto.ts (hunk 1 above)
Comment (abbiefarr) [reply backend/core/src/units-summary/dto/units-summary.dto.ts:6]: The `UnitsSummaryDto` doesn't extend `AbstractEntity` like `UnitTypeDto` does, and none of the fields are generated. So I _think_ we may not need to omit any fields and the file is ok like this??

**Review by abbiefarr — COMMENTED** [review]
> So on upsert, we would check if the user has selected to use individual units to compute the unit summary or to save the unit summary directly. Does that make sense to you?

Yes, that makes sense to me, I agree we shouldn't need a separate controller, since this can be a flag in the `listing` object (or we can repurposed the existing `disableUnitsAccordion` flag). 

Toggling between the two options may be disruptive, so we may also want to consider warnings when attempting that. E.g. if you've been using a summary but want to switch to individual units, and you don't have any data in your units table, that's pretty bad. So perhaps a warning that a data upload or refresh may be required? In the edit listing page (where I'm assuming this option is set), we could also display the contents of the table when you select each option as a visual warning too (our site current does this, with a mock summary table for now).

File: backend/core/src/units-summary/dto/units-summary.dto.ts (hunk 1 above)
Comment (seanmalbert) [reply backend/core/src/units-summary/dto/units-summary.dto.ts:6]: @abbiefarr , looks like you are right. I think we're good then!

**Comment by seanmalbert** [comment]
>Yes, that makes sense to me, I agree we shouldn't need a separate controller, since this can be a flag in the listing object (or we can repurposed the existing disableUnitsAccordion flag).

I think we can only repurpose it if we're sure that a user will never want to individually add units and disable the individual unit accordion. @slowbot , can you weigh in on this case?

> Toggling between the two options may be disruptive, so we may also want to consider warnings when attempting that. E.g. if you've been using a summary but want to switch to individual units, and you don't have any data in your units table, that's pretty bad. So perhaps a warning that a data upload or refresh may be required? In the edit listing page (where I'm assuming this option is set), we could also display the contents of the table when you select each option as a visual warning too (our site current does this, with a mock summary table for now).

I've discussed this with @slowbot and we agree that some type of warning message should display on the form. I think the plan was to start with helper text explaining that the user could incur data loss by switching between the options.

**Comment by seanmalbert** [comment]
@abbiefarr , I spun off #1633 to continue the discussion around handling the partner frontend use of the toggle.

**Review by seanmalbert — APPROVED** [review]

**Pull request merged**